### PR TITLE
update RootlessKit to v1.0.0

### PR DESCRIPTION
**Pull request by AkihiroSuda**
Only minor changes since v0.14.6, such as upgrading Go to 1.18:
https://github.com/rootless-containers/rootlesskit/compare/v0.14.6...v1.0.0

### CODE REVIEW

**Comment by thaJeztah** [comment]
Failure is unrelated;

```
=== RUN   TestDockerSuite/TestRunContainerWithRmFlagCannotStartContainer
    docker_cli_run_test.go:2770: Expected not to have containers 61a32bb2323e
        
    --- FAIL: TestDockerSuite/TestRunContainerWithRmFlagCannotStartContainer (2.75s)
```

**Comment by thaJeztah** [comment]
Looks like there's no changes in the vendored code from rootless kit itself, only changes in golang.org/x/sys 🤔

**Comment by AkihiroSuda** [comment]
> Looks like there's no changes in the vendored code from rootless kit itself, only changes in golang.org/x/sys 🤔

Yes, no substantial code change (https://github.com/rootless-containers/rootlesskit/compare/v0.14.6...v1.0.0).
I bumped the major version just because more than a year has passed since the release of v0.14.0 (and nearly 4 years since the initial commit) 😅

File: hack/dockerfile/install/rootlesskit.installer [hunk 1]
```diff
@@ -1,8 +1,8 @@
 #!/bin/sh
 
 # When updating, also update rootlesskit commit in vendor.conf accordingly
-# v0.14.6
-: "${ROOTLESSKIT_VERSION:=b25a0bad15d664c4ee8885ba622569425e918a68}"
+# v1.0.0
+: "${ROOTLESSKIT_VERSION:=1920341cd41e047834a21007424162a2dc946315}"
```
Comment (thaJeztah) [inline hack/dockerfile/install/rootlesskit.installer:5]: Wondering if we should continue to use a commit for this one, or also switch to use the version, as this is used as (e.g.)

```bash
go install .... github.com/rootless-containers/rootlesskit/cmd/.../@${ROOTLESSKIT_VERSION}`
```

We also need to change the `vendor.conf` to `vendor.mod` in the comment 😂

**Review by thaJeztah — COMMENTED** [review]
Given that the larger part of the changes are in `golang.org/x/sys`, would you mind doing a separate commit that updates that dependency? (I also left a comment about possibly switching from "commit" to "version" for `ROOTLESSKIT_VERSION`)

File: hack/dockerfile/install/rootlesskit.installer (hunk 1 above)
Comment (AkihiroSuda) [reply hack/dockerfile/install/rootlesskit.installer:5]: I suggest to continue using the hash, to minimize the risk of supply chain attacks.

**Comment by AkihiroSuda** [comment]
> Given that the larger part of the changes are in `golang.org/x/sys`, would you mind doing a separate commit that updates that dependency?

done

File: hack/dockerfile/install/rootlesskit.installer (hunk 1 above)
Comment (thaJeztah) [reply hack/dockerfile/install/rootlesskit.installer:5]: Yeah, it's not urgent, just that we'd already have the same in `vendor.mod` / `go.mod`

**Review by thaJeztah — APPROVED** [review]
LGTM

**Review by tianon — APPROVED** [review]

**Comment by thaJeztah** [comment]
Windows failure is `TestNetworkDBIslands`, which is (by now) famously flaky

**Pull request merged**